sql-streaming-akka/src/test/scala/org/apache/bahir/sql/streaming/akka/AkkaStreamSourceSuite.scala

```diff
@@ -0,0 +1,188 @@
+
+class AkkaStreamSourceSuite extends SparkFunSuite with SharedSparkContext with BeforeAndAfter {
+
+  protected var akkaTestUtils: AkkaTestUtils = _
+  protected val tempDir: File =
+    new File(System.getProperty("java.io.tmpdir") + "/spark-akka-persistence")
+
+  akkaTestUtils = new AkkaTestUtils
+  akkaTestUtils.setup()
+
+  before {
+    tempDir.mkdirs()
+  }
+
+  after {
+    Persistence.close()
+    BahirUtils.recursiveDeleteDir(tempDir)
+  }
+
+  protected val tmpDir: String = tempDir.getAbsolutePath
+
+  protected def createStreamingDataframe(dir: String = tmpDir): (SQLContext, DataFrame) = {
+
+    val sqlContext: SQLContext = new SQLContext(sc)
+
+    sqlContext.setConf("spark.sql.streaming.checkpointLocation", dir + "/checkpoint")
+
+    val dataFrame: DataFrame =
+      sqlContext.readStream.format("org.apache.bahir.sql.streaming.akka.AkkaStreamSourceProvider")
+        .option("urlOfPublisher", akkaTestUtils.getFeederActorUri())
+        .option("persistenceDirPath", dir + "/persistence").load()
+    (sqlContext, dataFrame)
+  }
+}
+
+class BasicAkkaSourceSuite extends AkkaStreamSourceSuite {
+
+  private def writeStreamResults(sqlContext: SQLContext, dataFrame: DataFrame,
+                                 waitDuration: Long): Boolean = {
+    import sqlContext.implicits._
+    dataFrame.as[(String, Timestamp)].writeStream.format("parquet")
+      .start(s"$tmpDir/parquet/t.parquet").awaitTermination(waitDuration)
+  }
+
+  private def readBackSreamingResults(sqlContext: SQLContext): mutable.Buffer[String] = {
+    import sqlContext.implicits._
+    val asList =
+      sqlContext.read.schema(AkkaStreamConstants.SCHEMA_DEFAULT)
+      .parquet(s"$tmpDir/parquet/t.parquet").as[(String, Timestamp)].map(_._1)
+      .collectAsList().asScala
+    asList
+  }
+
+  test("basic usage") {
+    val message = "Akka is a reactive framework"
+
+    akkaTestUtils.setMessage(message)
+    akkaTestUtils.setCountOfMessages(1)
+
+    val (sqlContext: SQLContext, dataFrame: DataFrame) = createStreamingDataframe()
+
+    writeStreamResults(sqlContext, dataFrame, 10000)
+
+    val resultBuffer: mutable.Buffer[String] = readBackSreamingResults(sqlContext)
+
+    assert(resultBuffer.size === 1)
+    assert(resultBuffer.head === message)
+  }
+
+  test("Send and receive 100 messages.") {
+    val message = "Akka is a reactive framework"
+
+    akkaTestUtils.setMessage(message)
+    akkaTestUtils.setCountOfMessages(100)
+
+    val (sqlContext: SQLContext, dataFrame: DataFrame) = createStreamingDataframe()
+
+    writeStreamResults(sqlContext, dataFrame, 10000)
+
+    val resultBuffer: mutable.Buffer[String] = readBackSreamingResults(sqlContext)
+
+    assert(resultBuffer.size === 100)
+    assert(resultBuffer.head === message)
+  }
+
+  test("params not provided") {
+    val persistenceDirPath = tempDir.getAbsolutePath + "/persistence"
+
+    val provider = new AkkaStreamSourceProvider
+    val sqlContext: SQLContext = new SQLContext(sc)
+
```

Comment:
Would you please change to use Sparksession instead of directly instantiating a SQLContext.